File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.py
class Solution:
    def sumEvenAfterQueries(self, nums: List[int], queries: List[List[int]]) -> List[int]:
        ans = 0
        for i in nums:
            if i % 2 == 0:
                ans += i
        res = []
        for val , idx in queries:
            if nums[idx] % 2== 0 and val % 2== 0:
                ans += val
            elif nums[idx] % 2 == 0 and val % 2 != 0:
                ans -= nums[idx]
            elif nums[idx] % 2 != 0 and val % 2 != 0:
                ans += nums[idx] + val
            nums[idx]+=val
            res.append(ans)
        return res
        

File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.java
class Solution {
    List<Integer> ans = new ArrayList<>();
    public int[] sumEvenAfterQueries(int[] nums, int[][] queries) {
        int res = 0;
        for (int i : nums){
            if (i % 2 == 0){
                res += i;
            }
        }
        for (int idx = 0 ; idx < queries.length; idx ++){
            int index = queries[idx][1];
            int val = queries[idx][0];
            if (nums[index] % 2 == 0 && val % 2 == 0){
                res += val;
            } else if (nums[index] % 2 == 0 && val % 2 != 0){
                res -= nums[index];
            } else if (nums[index] % 2 != 0 && val % 2 != 0){
                res += nums[index] + val;
            }       
            nums[index] += val;
            ans.add(res);
        }
        int[] a = new int[queries.length];
        int cnt = 0;
        for (int i : ans){
            a[cnt] = i;
            cnt ++;
        }
        return a;
    }
}